src/extensions/default/InAppNotifications/main.js

```diff
@@ -0,0 +1,372 @@
+                    if (tmpUrl !== localVersionInfoUrl) {
+                        $.ajax({
+                            url: tmpUrl,
+                            cache: false,
+                            type: "HEAD"
+                        }).fail(function (jqXHR, status, error) {
+                            localVersionInfoUrl = _getVersionInfoUrl("en");
+                        }).done(function (jqXHR, status, error) {
+                            localVersionInfoUrl = tmpUrl;
+                        }).always(function (jqXHR, status, error) {
+                            lookupPromise.resolve();
+                        });
+                    } else {
+                        localVersionInfoUrl = _getVersionInfoUrl("en");
+                        lookupPromise.resolve();
+                    }
+                }).done(function (jqXHR, status, error) {
+                    lookupPromise.resolve();
+                });
+            } else {
+                localVersionInfoUrl = _versionInfoUrl || _getVersionInfoUrl("en");
+                lookupPromise.resolve();
+            }
+
+            lookupPromise.done(function () {
+                $.ajax({
+                    url: localVersionInfoUrl,
+                    dataType: "json",
+                    cache: false
+                }).done(function (notificationInfo, textStatus, jqXHR) {
+                    lastInfoURLFetchTime = (new Date()).getTime();
+                    PreferencesManager.setViewState("lastNotificationURLFetchTime", lastInfoURLFetchTime);
+                    PreferencesManager.setViewState("notificationInfo", notificationInfo);
+                    result.resolve(notificationInfo);
+                }).fail(function (jqXHR, status, error) {
+                    // When loading data for unit tests, the error handler is
+                    // called but the responseText is valid. Try to use it here,
+                    // but *don't* save the results in prefs.
+
+                    if (!jqXHR.responseText) {
+                        // Text is NULL or empty string, reject().
+                        result.reject();
+                        return;
+                    }
+
+                    try {
+                        data = JSON.parse(jqXHR.responseText);
+                        result.resolve(data);
+                    } catch (e) {
+                        result.reject();
+                    }
+                });
+            });
+        } else {
+            result.resolve(data);
+        }
+
+        return result.promise();
+    }
+
+
+    /**
+     * Check for notifications, notification overlays are always displayed
+     *
+     * @return {$.Promise} jQuery Promise object that is resolved or rejected after the notification check is complete.
+     */
+    function checkForNotification() {
+        var result = new $.Deferred();
+        var versionInfoUrl;
+
+        _getNotificationInformation(versionInfoUrl)
```

Comment:
Why are we calling with `undefined`? Couldn't we just write `_getNotificationInformation()`?

Reply:
Need to add a parameter in the wrapper function which can be provided for testing.